src/main/scala/org/broadinstitute/dsde/workbench/sam/service/ResourceService.scala

```diff
@@ -68,15 +68,26 @@ class ResourceService(private val resourceTypes: Map[ResourceTypeName, ResourceT
     accessPolicyDAO.listAccessPolicies(resourceType.name, userInfo.userId)
   }
 
+  // IF Resource ID reuse is allowed (as defined by the Resource Type), then we can delete the resource
+  // ELSE Resource ID reuse is not allowed, and we enforce this by deleting all policies associated with the Resource,
+  //      but not the Resource itself, thereby orphaning the Resource so that it cannot be used or accessed anymore and
+  //      preventing a new Resource with the same ID from being created
   def deleteResource(resource: Resource): Future[Unit] = {
     for {
       policiesToDelete <- accessPolicyDAO.listAccessPolicies(resource)
       _ <- Future.traverse(policiesToDelete) {accessPolicyDAO.deletePolicy}
-      _ <- accessPolicyDAO.deleteResource(resource)
+      _ <- attemptDeleteResource(resource)
       _ <- Future.traverse(policiesToDelete) { policy => cloudExtensions.onGroupDelete(policy.email) }
     } yield ()
   }
 
+  private def attemptDeleteResource(resource: Resource): Future[Unit] = {
```

Comment:
I would call this maybeDeleteResource. "attempt" implies to me that we will try and possibly handle failures